### ISSUE TITLE
Improve the html report

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#5607]
I was answering at #5524 but I end up writing far too much so I'm moving out to a new issue. 

In my ideal world, on the html report, I would like to see all the issues that were found (I would also like to see the ones that were suppressed and why they were suppressed). Of course the suppressed ones should be a bit hidden so they don't mess up the default report.

Also, at the bottom of the report, also a bit hidden, I would like to have the list of active rules that report nothing. And even more hidden the disabled rules.

And the final thing is that, every rule should show how it was configured and from where that configuration comes from and a diff with the default values.

The problem with that is that we will need a lot of thinking about how to create a template with all that information and keep the report still easy. We don't want to mess up with the default use case of this report: check why detekt is angry again.

We could split this in little pieces because there are thing here that can not be implemented with our current core.

I think that a report like this could help a lot about why detekt is doing what it is doing. And it should help to spot mud configurations on the configuration.

**Comment by cortinico** [comment]
If someone is up for doing some UI mockups on Figma (or any other prototyping tool) this can be a great contribution <3

**Comment by lexa-diky** [comment]
Should we also use some templating library for html generation? Because even now HTML report generator looks ugly with all in-code templating going on. If we want to add more features it will become even more hard to keep track of what's going on.

I suggest using something like [FreeMarker](https://freemarker.apache.org) as it is seem to be the most popular one (I also like it the most personally 😄 )

3 issues that I see with this approach is:
1. additional library dependency - shouldn't be an issue if we choose something popular. Plus at the end there will not be a lot of work to migrate to something else if we want to for one or another reason.
2. bigger artifact - don't think that it does matter for detekt
3. templating libraries are often full of security vulnerabilities and people tend not to update detekt ever in their projects - do we really care for locally running app? Don't think so

**Comment by BraisGabin** [comment]
I'm not against but I think that before we do that we should know what we want and how we want it. And then decide how to implement it.

The good thing of the current implementation is that it's pure kotlin. You don't need to learn any new language. I know, templating is not that difficult but it is something you need to learn and I'm not sure if that will welcome more people to contribute or less.

**Comment by lexa-diky** [comment]
Just tried to migrate current report to freemarker. 

Code itself looks much nicer than current one, but I met a lot of issues with tooling. For some reason resource packing into JAR is super unstable for current gradle setup. Sometimes new templates just do not get into JAR which causes generation to crash. This behavior is unstable, sometimes it works, sometimes It does not. Will surely need to fix it before/if taking decision, not good developer experience.

**Comment by lexa-diky** [comment]
https://github.com/detekt/detekt/pull/5887 - just a showcase of how templates should look like. Supper messy, but you can get an idea of final version

**Comment by cortinico** [comment]
> https://github.com/detekt/detekt/pull/5887 - just a showcase of how templates should look like. Supper messy, but you can get an idea of final version

Thanks for sending this over @lexa-diky. I think this will improve the maintainability of the HTML report. I don't have a super strong opinion about FreeMarker, though, and I'm unsure if it's worth the extra technology that we'll add to our stack.

I'd like to hear other maintainers' opinion on this before we go forward (and I also want to make sure you don't invest too much time on this before we're not fully committed to invest on this).

**Comment by BraisGabin** [comment]
I don't think that our current implementation of `html` slow us down. I wouldn't add a new framework right now to the project.

If someone wants to bring to the table a full rewrite of the html report and s/he would like to do that with that technology I would be more than happy to go for it. But I don't feel that we should just change it.

I appreciate the PR and the show case lexa :)
